paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/memory/MemorySegmentAllocator.java

```diff
@@ -62,9 +62,8 @@ public MemorySegment allocate() {
             checkNotNull(segment, "Allocate null segment from memory manager for paimon.");
             checkArgument(segment.isOffHeap(), "Segment is not off heap from memory manager.");
             allocatedSegments.add(segment);
-            // TODO Use getOffHeapBuffer in MemorySegment after
-            // https://issues.apache.org/jira/browse/FLINK-32213
-            return MemorySegment.wrapOffHeapMemory((ByteBuffer) offHeapBufferField.get(segment));
+            ByteBuffer offHeapBuffer = segment.getOffHeapBuffer();
```

Comment:
Flink 1.15 and 1.16 and 1.17 will be break by this PR.

Reply:
Thanks for your remind,got it!